src/call_wrapper.ts

```diff
@@ -0,0 +1,106 @@
+import TranspilerHook from './transpiler_hook';
+import Module = WebAssembly.Module;
+
+class CallWrapper {
+
+    private static fillMemory(parameters: any[], writeableMemory: Uint32Array) {
+        // A copy of the parameters array is needed, to not override the content of the original one
+        const fixedParameters = parameters.concat();
+        let index = 0;
+
+        for (let i = 0; i < fixedParameters.length; i++) {
+            if (fixedParameters[i] instanceof Array) {
+                const array = fixedParameters[i];
+                writeableMemory[index++] = array.length;
+                fixedParameters[i] = index * 4;
+
+                for (const element of array) {
+                    writeableMemory[index++] = element;
+                }
+            }
+        }
+
+        return fixedParameters;
+    }
+
+    private static readMemory(parameters: any[], fixedParameters: any[], readableMemory: Uint32Array) {
+        let index = 0;
+
+        for (let i = 0; i < parameters.length; i++) {
+            if (parameters[i] instanceof Array) {
+                const array = parameters[i];
+
+                for (let j = 0; j < array.length; j++) {
+                    array[j] = readableMemory[fixedParameters[index] / 4 + j];
+                }
+
+                index++;
+            }
+        }
+    }
+
+    private static calculateInitialMemorySize(params: any[]): number {
+        const pageSize = Math.pow(2, 16);
+        const memoryElementCount = params
+            .filter((parameter) => parameter instanceof Array)
+            .map((array) => array.length + 1)
+            .reduce((accumulator, current) => accumulator + current, 0);
+
+        return Math.ceil((memoryElementCount * 4) / pageSize);
+    }
+
+    private readonly hook: TranspilerHook;
+    private readonly wasmModule: Module;
+
+    private callFunctionName: string;
```

Comment:
`callFunctionName` is probably not the best name here. Maybe just `functionName`?